GalaxyZooTouchTable/GalaxyZooTouchTable/ViewModels/SpaceViewModel.cs

```diff
@@ -30,6 +31,34 @@ public string SpaceCutoutUrl
         public SpaceViewModel()
         {
             PrepareForNewPosition();
+            Messenger.Default.Register<ClassificationRingNotifier>(this, OnGalaxyInteraction);
+        }
+
+        private void OnGalaxyInteraction(ClassificationRingNotifier RingNotifier)
+        {
+            foreach (TableSubject SpaceViewGalaxy in CurrentGalaxies)
+            {
+                if (RingNotifier.SubjectId == SpaceViewGalaxy.Subject.Id)
+                {
+                    switch (RingNotifier.Status)
+                    {
+                        case RingNotifierStatus.IsCreating:
+                            SpaceViewGalaxy.AddRing(RingNotifier.User);
+                            break;
+                        case RingNotifierStatus.IsSubmitting:
+                            SpaceViewGalaxy.DimRing(RingNotifier.User);
+                            break;
+                        case RingNotifierStatus.IsHelping:
+                            SpaceViewGalaxy.RemoveRing(RingNotifier.User);
+                            break;
+                        default:
+                            break;
+                    }
+                } else if (RingNotifier.Status == RingNotifierStatus.IsLeaving)
+                {
+                    SpaceViewGalaxy.RemoveRing(RingNotifier.User);
```

Comment:
This logic seems a little weird. Why is `IsLeaving` not checked for in the switch?

Reply:
Because of the check for `Ring.Notifier.SubjectId == SpaceViewGalaxy.Subject.Id`, which checks that the current subject a user is classifying matches the correct subject on the board.

Instead, when a user is leaving the table, _all_ of their rings should be removed, including those subjects they have already seen and classified.